File: frontend/src/modules/graph/application/use-cases/stageAll.ts
import { WorkingChangesWriter } from "@modules/graph/application/contracts/WorkingChangesWriter";

export async function stageAll(
  workingChangesWriter: WorkingChangesWriter,
  repositoryPath: string
): Promise<void> {
  return workingChangesWriter.stageAll(repositoryPath);
}

File: backend/src/app/working_changes/stage_all.rs
use std::path::Path;

use crate::app::working_changes::contracts::WorkingChangesWriter;
use crate::domain::errors::AppError;

pub fn stage_all(
    working_changes_writer: &dyn WorkingChangesWriter,
    repository_path: &Path,
) -> Result<(), AppError> {
    working_changes_writer.stage_all(repository_path)
}
